### PR TITLE
flow diagram nodes size

### LINKED ISSUE #347: Allow users to specify width for nodes in flow diagram
If users want conformity in size between shapes in their diagrams there is not a good way to do that today. They must resort to hacky and mostly ineffective workarounds like inserting zero-width spaces into the label names.

It would be better if users could optionally set the dimensions of their nodes.

### PR DESCRIPTION
**Pull request by MykolaGolubyev**
image

Closes #347

### CODE REVIEW

File: znai-diagrams/src/main/java/com/twosigma/znai/diagrams/graphviz/gen/GraphvizFromJsonGen.java [hunk 1]
```diff
@@ -137,7 +157,17 @@ private String generateNodes(Collection<DiagramNode> nodes) {
...
+        return node.getId() + " [label=\"" + generateNodeLabel(node) + "\"" +
+                generateSizeAttributesIfRequired(node) + "];";
+    }
+
+    private String generateSizeAttributesIfRequired(DiagramNode node) {
+        if (!nodesConfig.isSizeDefined(node)) {
+            return "";
+        }
+
+        return " fixedsize=true" + (nodesConfig.isWidthDefined(node) ? " width=" + nodesConfig.getWidth(node) : "") +
```
Comment (byc1234) [inline znai-diagrams/src/main/java/com/twosigma/znai/diagrams/graphviz/gen/GraphvizFromJsonGen.java]: Let's use StringBuilder to make the code more readable.

    StringBuilder sb = new StringBuilder(" fixedsize=true");
    if (nodesConfig.isWidthDefined(node) {
        sb.append(" width=" + nodesConfig.getWidth(node));
    }
    if (nodesConfig.isHeightDefined(node) {
        sb.append(" height=" + nodesConfig.getHeight(node));
    }
    return sb.toString();

**Comment by byc1234** [comment]
Could you briefly explain how graphviz works together with znai?

File: znai-diagrams/src/main/java/com/twosigma/znai/diagrams/graphviz/gen/GraphvizFromJsonGen.java (hunk 1 above)
Comment (MykolaGolubyev) [reply znai-diagrams/src/main/java/com/twosigma/znai/diagrams/graphviz/gen/GraphvizFromJsonGen.java]: done

**Comment by MykolaGolubyev** [comment]
@byc1234 we interact with graphviz using command line interface. We generate a graphviz input and then send it to graphviz external process. The resulting SVG is handled on JavaScript side by processing SVG and making some adjustments.

**Pull request merged**